### ISSUE TITLE
[Rattle GTK] `cairoConversion.c: error: expected ';' before string constant`

### ISSUE THREAD

**Issue by barracuda156** [gjwgit/rattleng#61]
@gjwgit Is there a working version of `RGtk2` around? I tried to build the last one from CRAN (I think your website has the same), it fails both with gtk3 and gtk2:

```
In file included from RGtk2/cairo.h:7,
                 from cairoConversion.c:1:
./RGtk2/gobject.h:375:1: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
  375 | GTimeVal* asCGTimeVal(USER_OBJECT_ s_timeval);
      | ^~~~~~~~
In file included from /opt/local/include/glib-2.0/glib/galloca.h:34,
                 from /opt/local/include/glib-2.0/glib.h:32,
                 from ./RGtk2/gobject.h:7:
/opt/local/include/glib-2.0/glib/gtypes.h:580:8: note: declared here
  580 | struct _GTimeVal
      |        ^~~~~~~~~
./RGtk2/gobject.h:376:1: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
  376 | USER_OBJECT_ asRGTimeVal(const GTimeVal *timeval);
      | ^~~~~~~~~~~~
/opt/local/include/glib-2.0/glib/gtypes.h:580:8: note: declared here
  580 | struct _GTimeVal
      |        ^~~~~~~~~
cairoConversion.c: In function 'asRCairoPath':
cairoConversion.c:45:25: error: 'PROBLEM' undeclared (first use in this function)
   45 |                         PROBLEM "Converting Cairo path: did not understand type %d", data->header.type
      |                         ^~~~~~~
cairoConversion.c:45:25: note: each undeclared identifier is reported only once for each function it appears in
cairoConversion.c:45:32: error: expected ';' before string constant
   45 |                         PROBLEM "Converting Cairo path: did not understand type %d", data->header.type
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                ;
cairoConversion.c: In function 'asCCairoPath':
cairoConversion.c:95:33: error: 'PROBLEM' undeclared (first use in this function)
   95 |                                 PROBLEM "Converting Cairo path: did not understand type %d", type
      |                                 ^~~~~~~
cairoConversion.c:95:40: error: expected ';' before string constant
   95 |                                 PROBLEM "Converting Cairo path: did not understand type %d", type
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                        ;
make: *** [cairoConversion.o] Error 1
ERROR: compilation failed for package ‘RGtk2’
```
This is with gcc13 on 10.6.

P. S. Unfortunately, the new version won’t work: we do not have that exotic language for Darwin ppc.

**Comment by barracuda156** [comment]
Oh, and `cairoDevice` fails likewise:
```
/opt/local/bin/gcc-mp-13 -I"/opt/local/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I/opt/local/include/gtk-2.0 -I/opt/local/lib/gtk-2.0/include -I/opt/local/include/pango-1.0 -I/opt/local/include/gio-unix-2.0 -I/opt/local/include/cairo -I/opt/local/include/atk-1.0 -I/opt/local/include/gdk-pixbuf-2.0 -I/opt/local/include/pango-1.0 -I/opt/local/include/harfbuzz -I/opt/local/include/pango-1.0 -I/opt/local/include/fribidi -I/opt/local/include/cairo -I/opt/local/include/pixman-1 -I/opt/local/include/harfbuzz -I/opt/local/include/glib-2.0 -I/opt/local/lib/glib-2.0/include -I/opt/local/include/freetype2 -I/opt/local/include/libpng16 -DR_LEGACY_S_DEFS=1  -isystem/opt/local/include/LegacySupport -I/opt/local/include    -fPIC  -pipe -Os -arch ppc  -c cairoDevice.c -o cairoDevice.o
In file included from /opt/local/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /opt/local/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /opt/local/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /opt/local/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /opt/local/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /opt/local/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /opt/local/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /opt/local/include/gtk-2.0/gtk/gtk.h:33,
                 from cairoDevice.h:1,
                 from cairoDevice.c:1:
/opt/local/include/gtk-2.0/gtk/gtktypeutils.h:236:1: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      | ^~~~
In file included from /opt/local/include/glib-2.0/gobject/gobject.h:26,
                 from /opt/local/include/glib-2.0/gobject/gbinding.h:31,
                 from /opt/local/include/glib-2.0/glib-object.h:24,
                 from /opt/local/include/glib-2.0/gio/gioenums.h:30,
                 from /opt/local/include/glib-2.0/gio/giotypes.h:30,
                 from /opt/local/include/glib-2.0/gio/gio.h:28,
                 from /opt/local/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /opt/local/include/gtk-2.0/gdk/gdk.h:32,
                 from /opt/local/include/gtk-2.0/gtk/gtk.h:32:
/opt/local/include/glib-2.0/gobject/gtype.h:723:1: note: declared here
  723 | {
      | ^
In file included from /opt/local/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /opt/local/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /opt/local/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /opt/local/include/gtk-2.0/gtk/gtk.h:126:
/opt/local/include/gtk-2.0/gtk/gtktooltips.h:73:3: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |   ^~~~~~~~
In file included from /opt/local/include/glib-2.0/glib/galloca.h:34,
                 from /opt/local/include/glib-2.0/glib.h:32,
                 from /opt/local/include/glib-2.0/gobject/gbinding.h:30:
/opt/local/include/glib-2.0/glib/gtypes.h:580:8: note: declared here
  580 | struct _GTimeVal
      |        ^~~~~~~~~
cairoDevice.c: In function 'initCairoDevice':
cairoDevice.c:1506:7: error: 'PROBLEM' undeclared (first use in this function)
 1506 |       PROBLEM  "unable to start device cairo" ERROR;
      |       ^~~~~~~
cairoDevice.c:1506:7: note: each undeclared identifier is reported only once for each function it appears in
cairoDevice.c:1506:14: error: expected ';' before string constant
 1506 |       PROBLEM  "unable to start device cairo" ERROR;
      |              ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |              ;
cairoDevice.c: In function 'do_Cairo':
cairoDevice.c:1532:5: error: 'PROBLEM' undeclared (first use in this function)
 1532 |     PROBLEM "Cairo device: invalid width or height" ERROR;
      |     ^~~~~~~
cairoDevice.c:1532:12: error: expected ';' before string constant
 1532 |     PROBLEM "Cairo device: invalid width or height" ERROR;
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |            ;
make: *** [cairoDevice.o] Error 1
ERROR: compilation failed for package ‘cairoDevice’
```

**Comment by gjwgit** [comment]
Unfortunately the latest I have is as on https://rattle.togaware.com RGtk2_2.20.36.2

Testing on Ubuntu and Windows 10 seems okay. But not having a Mac I haven't any recent tests there.

It's rather awkward since CRAN make the decisions. I was in touch with the RGtk2 author but he was not able to get it restored to CRAN. So I don't really have anywhere to go with the GTK version at this time, unless someone is able to resurrect it.

**Comment by barracuda156** [comment]
@gjwgit Does the author have a GitHub? We do need CRAN at all, tbh. And their policies are suboptimal, to put it politely.

**Comment by barracuda156** [comment]
@gjwgit Perhaps not worth opening a separate issue, so just writing here: `rattle` is now in Macports: https://ports.macports.org/port/R-rattle
